aicssegmentation/structure_wrapper/WorkflowEngine.py

```diff
@@ -0,0 +1,141 @@
+from aicssegmentation.structure_wrapper_config.structure_config_utils import load_workflow_config, parse_config_to_objects
+import numpy as np
+from aicssegmentation.structure_wrapper.WorkflowStep import WorkflowStep
+from typing import List
+
+
+class WorkflowEngine:
+    """
+    A class to define a whole aics-segmentation workflow
+    """
+    def __init__(self, workflow_name: str, image: np.ndarray):
+        """
+        Constructor for the WorkflowEngine object
+
+        Params:
+            workflow_name (str): dictionary object containing information about this workflow step
+            image (np.ndarray):  image to perform workflow on
+        """
+        self.workflow_name: str = workflow_name   # Workflow name
+        self.steps: list = self.__get_steps()        # List of WorkflowSteps for this workflow
+        self.next_step: int = 0                 # Next step to execute
+        self.starting_image: np.ndarray = image          # Initial image
+
+    def __get_steps(self) -> List:
```

Comment:
Full typing annotation for a typed list would be `List[WorkflowStep]`

Also, I would rename this method to something more explicit like `__load_steps_from_config` because we're going to need a public `get_steps` method or a `steps` property (that I don't see currently implemented) that the plugin would use to then display the steps on the UI.